unit-tests/unit-test-config.py

```diff
@@ -139,16 +139,37 @@ def find_includes( filepath ):
     return filelist
 
 def process_cpp( dir, builddir ):
+    global regex, required_tags, list_only, available_tags, available_tests
     found = []
     shareds = []
     statics = []
-    for f in find( dir, '(^|/)test-.*\.cpp$' ):
+    if regex:
+        pattern = re.compile( regex )
+    for f in file.find( dir, '(^|/)test-.*\.cpp$' ):
         testdir = os.path.splitext( f )[0]                          # "log/internal/test-all"  <-  "log/internal/test-all.cpp"
         testparent = os.path.dirname(testdir)                       # "log/internal"
         # Each CMakeLists.txt sits in its own directory
-        os.makedirs( builddir + '/' + testdir, exist_ok = True );   # "build/log/internal/test-all"
+        os.makedirs( builddir + '/' + testdir, exist_ok = True )    # "build/log/internal/test-all"
```

Comment:
Shouldn't happen with --list or if the test doesn't match

Reply:
Moved to after all the checks